app/portainer/helpers/webhookHelper.js

```diff
@@ -0,0 +1,12 @@
+angular.module('portainer.app')
+.factory('WebhookHelper', ['$location', 'API_ENDPOINT_WEBHOOKS', function WebhookHelperFactory($location,API_ENDPOINT_WEBHOOKS) {
+  'use strict';
+  var helper = {};
+
+  helper.returnWebhookUrl = function(token) {
+    return $location.protocol() + "://" +  $location.host() + ":"
+    + $location.port() + "/" + API_ENDPOINT_WEBHOOKS + "/" + token;
```

Comment:
Only include `:<port>` in the URL if it is not the default port of the protocol (i.e. not 80 for HTTP and not 443 for HTTPS). A URL like `https://portainer.example.com:443/api/webhooks/12345...` does not look good.

Reply:
Agree

Reply:
Below is an untested code, but something along this line should do the trick.

```js
helper.returnWebhookUrl = function(token) {
  var displayPort = $location.protocol().toLowerCase() == "http" && $location.port() == 80 || $location.protocol().toLowerCase() == "https" && $location.port() == 443 ? "" : ":" + $location.port();
  return $location.protocol() + "://" +  $location.host() + displayPort + "/" + API_ENDPOINT_WEBHOOKS + "/" + token;
};
```